simphony/io/tests/test_file_lattice.py

```diff
@@ -0,0 +1,180 @@
+import os
+import tempfile
+import shutil
+import unittest
+import tables
+import numpy
+
+from simphony.cuds.lattice import Lattice
+from simphony.cuds.lattice import LatticeNode
+from simphony.io.file_lattice import FileLattice
+
+from simphony.core.cuba import CUBA
+
+from simphony.io.h5_cuds import H5CUDS
+
+
+class CustomRecord(tables.IsDescription):
+
+    class data(tables.IsDescription):
+
+        material_id = tables.Int32Col(pos=0)
+        velocity = tables.Float64Col(pos=1, shape=3)
+        density = tables.Float64Col(pos=2)
+
+    mask = tables.BoolCol(pos=1, shape=(3,))
+
+
+class TestFileLattice(unittest.TestCase):
+
+    def setUp(self):
+
+        self.temp_dir = tempfile.mkdtemp()
+
+        self.filename = os.path.join(self.temp_dir, 'test_file.cuds')
+        self.file = H5CUDS.open(self.filename, mode='w')
+
+        self.lattice = Lattice('test_lattice', 'cubic', (1.0, 1.0, 1.0),
+                               (4, 5, 6), (0.0, 0.0, 0.0))
+
+        for i in xrange(4):
+            for j in xrange(5):
+                for k in xrange(6):
+                    N = LatticeNode((i, j, k), {CUBA.MATERIAL_ID: 1})
+                    self.lattice.update_node(N)
+
+        for i in xrange(2):
+            for j in xrange(3):
+                for k in xrange(6):
+                    N = LatticeNode((i+1, j+1, k),
+                                    {CUBA.MATERIAL_ID: 0,
+                                     CUBA.DENSITY: 1.0,
+                                     CUBA.VELOCITY: [0.0, 0.0, 1.0]})
+                    self.lattice.update_node(N)
+
+        lat = self.lattice
+        self.filelattice = FileLattice(self.file._handle, lat.name, lat.type,
+                                       lat.base_vect, lat.size, lat.origin,
+                                       CustomRecord)
+        for N in lat.iter_nodes():
+            self.filelattice.update_node(N)
+
+    def tearDown(self):
+        if os.path.exists(self.filename):
+            self.file.close()
+        shutil.rmtree(self.temp_dir)
+
+    def test_file_lattice_constructed_from_Lattice(self):
+        """ Checks that FileLattice constructed from Lattice with a
+        CustomRecord column description has correct attributes
+
+        """
+        self.assertEqual(self.filelattice.name, 'test_lattice')
+        self.assertEqual(self.filelattice.type, 'cubic')
+        self.assertItemsEqual(self.filelattice.base_vect,
```

Comment:
Thanks for the tip. I'm now using assert_array_equal (without the s) for all numpy array checks.